core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1005,7 +1073,33 @@ public void commitBlock(long workerId, long usedBytesOnTier, String tierAlias,
     MasterWorkerInfo worker = mWorkers.getFirstByField(ID_INDEX, workerId);
     // TODO(peis): Check lost workers as well.
     if (worker == null) {
-      throw new NotFoundException(ExceptionMessage.NO_WORKER_FOUND.getMessage(workerId));
+      /*
+       * If the worker is not recognized:
+       * 1. [Probably] The worker has been decommissioned and removed from the active worker list
+       * 2. [Possible] The worker has not finished its register process. Maybe the master has
+       *    failed over and the worker has not registered to this new primary.
+       * 3. [Unlikely] The worker does not belong to this cluster and has never registered.
+       *    This is unlikely because the worker has an ID and it must be from some master.
+       * 4. [Unlikely] The worker is lost to the master. This is unlikely because the CommitBlock
+       *    call is from the worker. This is more possibly the master is busy and did not
+       *    handle the worker's heartbeat message for too long.
+       */
+      worker = mDecommissionedWorkers.getFirstByField(ID_INDEX, workerId);
+      if (worker == null) {
+        throw new NotFoundException(ExceptionMessage.NO_WORKER_FOUND.getMessage(workerId));
```

Comment:
Out of scope of this PR, but I find it curious that lost workers are not allowed to commit blocks. Does worker-side code knows about this and maybe do a re-registration before retrying committing?

Reply:
Good question.

Most likely a lost worker will not commit a block (otherwise why is it lost?)

If it's because a network separation then we don't really have a good definition for that across the current Alluxio architecture.

One other possible reason is the master is flooded and could not process the worker heartbeat in time (the worker is working fine). In that case we deem the master state is not health and the admin should take a look (and decommission a worker is less important now).